packages/block-editor/src/components/media-upload/index.native.js

```diff
@@ -137,15 +143,27 @@ export class MediaUpload extends Component {
 		const {
 			allowedTypes = [],
 			__experimentalOnlyMediaLibrary,
+			isAudioBlockMediaUploadEnabled,
 		} = this.props;
 
 		return this.getAllSources()
 			.filter( ( source ) => {
-				return __experimentalOnlyMediaLibrary
-					? source.mediaLibrary
-					: allowedTypes.some( ( allowedType ) =>
+				if ( __experimentalOnlyMediaLibrary ) {
+					return source.mediaLibrary;
+				} else if (
+					allowedTypes.every(
+						( allowedType ) =>
+							allowedType === MEDIA_TYPE_AUDIO &&
```

Comment:
I see that this calculation is not specific to the loop so probably we could move it to a constant defined before the `return` line. This way we prevent running the loop that is done when calling the `every` function in all the passes of the `filter` loop, wdyt?

Reply:
Thanks for the feedback. I am not quite clear on how I would be able to do this without causing an issue since `source.types.includes( allowedType )` is utilized to verify that the source supports the specified allowedType. Re-reading to see if I am missing the context.

Reply:
I think before I added 245a895 we would be able to extract this into a constant of its own but in this case, it might not be optimal.

Reply:
Oh sorry, I overlooked the `source.types.includes( allowedType )` part of the condition. In this case, as you mentioned, we can't extract it as it requires `source`.

Reply:
Indeed! Thanks for confirming 👍🏾